pkg/archive/archive_unix.go

```diff
@@ -6,11 +6,27 @@ import (
 	"archive/tar"
 	"errors"
 	"os"
+	"path/filepath"
+	"strings"
 	"syscall"
 
 	"github.com/docker/docker/pkg/system"
 )
 
+// fixVolumePathPrefix does platform specific processing to ensure that if
+// the path being passed in is not in a volume path format, convert it to one.
+func fixVolumePathPrefix(srcPath string) string {
+	return srcPath
+}
+
+// getWalkRoot calculates the root path when performing a TarWithOptions.
+// We use a seperate function as this is platform specific. On Linux, we
+// can't use filepath.Join(srcPath,include) because this will clean away
+// a trailing "." or "/" which may be important.
+func getWalkRoot(srcPath string, include string) string {
+	return strings.Join([]string{srcPath, include}, string(filepath.Separator))
```

Comment:
in this case I believe it's simpler to do `return srcPath + string(os.Separator) + include`